cmd/createtree/pem.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/google/trillian/crypto/keys"
+	"github.com/google/trillian/crypto/keyspb"
+)
+
+var (
+	pemKeyPath = flag.String("pem_key_path", "", "Path to the private key PEM file")
+	pemKeyPass = flag.String("pem_key_password", "", "Password of the private key PEM file")
+)
+
+func init() {
+	keyHandlers["PEMKeyFile"] = func() (proto.Message, error) {
+		if *pemKeyPath == "" {
+			return nil, errors.New("empty pem_key_path")
+		}
+		if *pemKeyPass == "" {
+			return nil, fmt.Errorf("empty password for PEM key file %q", *pemKeyPath)
+		}
+
+		return &keyspb.PEMKeyFile{
+			Path:     *pemKeyPath,
+			Password: *pemKeyPass,
+		}, nil
+	}
+
+	keyHandlers["PrivateKey"] = func() (proto.Message, error) {
```

Comment:
Consider making each of these anonymous functions real functions for ease on the eye and ease of testing.